commit 992a773cb9bbd17e79647a3bfced86f2a4c8d5ad
Author: Prabhath Sajeepa <psajeepa@purestorage.com>
Date:   Wed Nov 28 11:11:29 2018 -0700

    nvme-rdma: fix double freeing of async event data
    
    [ Upstream commit 6344d02dc8f886b6bbcd922ae1a17e4a41500f2d ]
    
    Some error paths in configuration of admin queue free data buffer
    associated with async request SQE without resetting the data buffer
    pointer to NULL, This buffer is also freed up again if the controller
    is shutdown or reset.
    
    Signed-off-by: Prabhath Sajeepa <psajeepa@purestorage.com>
    Reviewed-by: Roland Dreier <roland@purestorage.com>
    Signed-off-by: Christoph Hellwig <hch@lst.de>
    Signed-off-by: Sasha Levin <sashal@kernel.org>

diff --git a/drivers/nvme/host/rdma.c b/drivers/nvme/host/rdma.c
index dc042017c293..b6a28de682e8 100644
--- a/drivers/nvme/host/rdma.c
+++ b/drivers/nvme/host/rdma.c
@@ -184,6 +184,7 @@ static int nvme_rdma_alloc_qe(struct ib_device *ibdev, struct nvme_rdma_qe *qe,
 	qe->dma = ib_dma_map_single(ibdev, qe->data, capsule_size, dir);
 	if (ib_dma_mapping_error(ibdev, qe->dma)) {
 		kfree(qe->data);
+		qe->data = NULL;
 		return -ENOMEM;
 	}
 
@@ -816,6 +817,7 @@ static int nvme_rdma_configure_admin_queue(struct nvme_rdma_ctrl *ctrl,
 out_free_async_qe:
 	nvme_rdma_free_qe(ctrl->device->dev, &ctrl->async_event_sqe,
 		sizeof(struct nvme_command), DMA_TO_DEVICE);
+	ctrl->async_event_sqe.data = NULL;
 out_free_queue:
 	nvme_rdma_free_queue(&ctrl->queues[0]);
 	return error;
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1121_992a773cb9bbd17e79647a3bfced86f2a4c8d5ad.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1121_992a773cb9bbd17e79647a3bfced86f2a4c8d5ad.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
